File: Cpp/3151_special_array_i.cpp
// https://leetcode.cn/problems/special-array-i/

class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        int pre = nums[0] % 2;
        for (int i = 1; i < nums.size(); ++i) {
            int cur = nums[i] % 2;
            if (pre + cur != 1) {
                return false;
            }
            pre = cur;
        }
        return true;
    }
};

File: Python3/3151_special_array_i.py
# https://leetcode.cn/problems/special-array-i/

class Solution:
    def isArraySpecial(self, nums):
        """
        Inputs:
            nums: list[int]
        Outputs:
            res: bool
        """
        pre = nums[0] % 2
        for i in range(1, len(nums)):
            cur = nums[i] % 2
            if pre + cur != 1:
                return False
            pre = cur
        return True